### ISSUE TITLE
Clean code pt.2: keep it DRY

### ISSUE THREAD

**Issue by kresimir-coko** [Rinakat225/Todo-list#30]
In NewTaskForm.jsx we have a situation where we are resetting state twice using the same 4 lines. In [one example](https://github.com/Rinakat225/Todo-list/blob/master/src/mainComponents/NewTaskForm.jsx#L32-L38) those 4 lines are separated by an empty line, in the [other](https://github.com/Rinakat225/Todo-list/blob/master/src/mainComponents/NewTaskForm.jsx#L32-L38) they are not.

<img width="685" alt="image" src="https://github.com/Rinakat225/Todo-list/assets/24564752/9664d6da-3ddb-4b8b-8dbe-6b97c94cc98b">

TLDR: Extract the 4 lines into a `resetState` function.
